sdv/constraints/base.py

```diff
@@ -112,31 +121,72 @@ def __init__(self, handling_strategy):
+        """Fit ``Constraint`` class to data.
+
+        If ``disable_columns_model`` is False, then this method will fit
+        a ``GaussianCopula`` model to the relevant columns in ``table_data``.
+        Subclasses can overwrite this method, or overwrite the ``_fit`` method
+        if they will not be needing the model to handle conditional sampling.
 
         Args:
             table_data (pandas.DataFrame):
                 Table data.
         """
-        del table_data
+        if not self.disable_columns_model:
+            data_to_model = table_data[list(self.constraint_columns)]
+            self._hyper_transformer = HyperTransformer(dtype_transformers={
+                'O': 'one_hot_encoding',
+            })
+            transformed_data = self._hyper_transformer.fit_transform(data_to_model)
+            self._columns_model = GaussianMultivariate()
+            self._columns_model.fit(transformed_data)
+
+        return self._fit(table_data)
 
     def _transform(self, table_data):
         return table_data
 
     def _validate_constraint_columns(self, table_data):
         """Validate the columns in ``table_data``.
 
-        If any columns in ``constraint_columns`` are not present in ``table_data``,
-        this method will raise a ``MissingConstraintColumnError``.
+        If ``disable_columns_model`` is True and any columns in ``constraint_columns``
+        are not present in ``table_data``, this method will raise a
+        ``MissingConstraintColumnError``. Otherwise it will return the ``table_data``
+        unchanged. If ``disable_columns_model`` is False, then this method will sample
+        any missing ``constraint_columns`` from its model conditioned on the
+        ``constraint_columns`` that ``table_data`` does contain. If ``table_data``
+        doesn't contain any of the ``constraint_columns`` then a
+        ``MissingConstraintColumnError`` will be raised.
 
         Args:
             table_data (pandas.DataFrame):
                 Table data.
         """
-        if any(col not in table_data.columns for col in self.constraint_columns):
+        if self.disable_columns_model:
+            if any(col not in table_data.columns for col in self.constraint_columns):
+                raise MissingConstraintColumnError()
+            return table_data
+
+        missing_columns = [col for col in self.constraint_columns if col not in table_data.columns]
+
+        if len(missing_columns) == 0:
+            return table_data
+        if all(col in missing_columns for col in self.constraint_columns):
             raise MissingConstraintColumnError()
 
+        condition_columns = [c for c in self.constraint_columns if c in table_data.columns]
+        unrelated_columns = [c for c in table_data.columns if c not in self.constraint_columns]
+        conditions = self._hyper_transformer.transform(table_data[condition_columns])
+        sampled_data = conditions.apply(lambda d: self._columns_model.sample(
```

Comment:
I would do this in `for` loop instead of using an `apply/lambda`, and also use a `groupby` on the existing condition columns to avoid making a call to `sample` for each line.